src/core/tFunctions.ml

```diff
@@ -610,11 +610,23 @@ let concat e1 e2 =
 	) in
 	mk e e2.etype (punion e1.epos e2.epos)
 
+let extract_param_type (_,t,_) = t
+let extract_param_types = List.map extract_param_type
+let extract_param_name (n,_,_) = n
+let lookup_param n l =
+	let rec loop l = match l with
```

Comment:
`lookup_param` could be made `rec` instead of declaring additional inner function.

Reply:
It could, but I think this is actually more OCaml-idiomatic.